### PR TITLE
[stdlib] Some cleanup enabled by _alwaysEmitIntoClient.

### PR DESCRIPTION
**Pull request by milseman**
Refactor some copy-pasted code into a helper computed variable and
outline some cold paths.

### CODE REVIEW

**Comment by milseman** [comment]
@swift-ci please test

File: stdlib/public/core/Character.swift [hunk 1]
```diff
@@ -83,8 +83,7 @@ extension Character {
     _internalInvariant(_str._guts.isFastUTF8)
 
     // TODO(@eject): Switch to a helper property on StringObject/StringGuts.
```
Comment (slavapestov) [inline stdlib/public/core/Character.swift]: You can remove this TODO and the one below no?

**Review by slavapestov — APPROVED** [review]

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please benchmark

**Pull request merged**